src/hidimstat/dcrt.py

```diff
@@ -11,265 +11,379 @@
+        results = Parallel(n_jobs, verbose=joblib_verbose)(
+            delayed(_lasso_distillation_residual)(
+                X_,
+                y_,
+                idx,
+                coef_full=coef_X_full,
+                sigma_X=sigma_X,
+                cv=cv,
+                alpha=alpha,
+                alpha_max_fraction=alpha_max_fraction,
+                n_alphas=n_alphas,
+                alphas=alphas,
+                fit_y=fit_y,
+                n_jobs=1,  # the function is already called in parallel
+                random_state=random_state,
             )
-        else:
-            raise ValueError(f"{loss} loss is not supported.")
+            for idx in selection_set
+        )
     elif statistic == "randomforest":
-        if loss == "least_square":
-            results = Parallel(n_jobs, verbose=joblib_verbose)(
-                delayed(_rf_distillation)(
-                    X,
-                    y,
-                    idx,
-                    Sigma_X=Sigma_X,
-                    cv=3,
-                    use_cv=use_cv,
-                    alpha=alpha,
-                    n_jobs=1,
-                    n_alphas=n_alphas,
-                    ntree=ntree,
-                    loss=loss,
-                    problem_type=problem_type,
-                    random_state=random_state,
-                )
-                for idx in selection_set
+        # For distillation of X use least_square loss
+        results = Parallel(n_jobs, verbose=joblib_verbose)(
+            delayed(_rf_distillation)(
+                X_,
+                y_,
+                idx,
+                sigma_X=sigma_X,
+                cv=cv,
+                alpha=alpha,
+                alpha_max_fraction=alpha_max_fraction,
+                n_alphas=n_alphas,
+                alphas=alphas,
+                ntree=ntree,
+                problem_type=problem_type,
+                n_jobs=1,  # the function is already called in parallel
+                random_state=random_state,
             )
-        else:
-            raise ValueError(f"{loss} loss is not supported.")
+            for idx in selection_set
+        )
     else:
         raise ValueError(f"{statistic} statistic is not supported.")
-    ts = np.zeros(n_features)
-    ts[selection_set] = np.array([i for i in results])
+
+    # contatenate result
+    selection_features = np.ones((n_features,), dtype=bool)
+    selection_features[non_selection] = 0
+    X_res = np.array([i[0] for i in results])
+    sigma2_X = np.array([i[1] for i in results])
+    y_res = np.array([i[2] for i in results])
+    return selection_features, X_res, sigma2_X, y_res
+
+
+def dcrt_pvalue(
+    selection_features,
+    X_res,
+    sigma2_X,
+    y_res,
+    fdr=0.1,
+    fdr_control="bhq",
+    reshaping_function=None,
+    scaled_statistics=False,
+):
+    """
+    Calculate p-values and identify significant features using the dCRT test statistics.
```

Comment:
I'm quite unsure why we need to separate it from the previous. Are there cases where we would use one but not he other ?

Reply:
see my response above

Reply:
See issue #201.